src/pages/VenuePage/VenuePage.tsx

```diff
@@ -204,18 +202,18 @@ export const VenuePage: React.FC = () => {
 
   // useVenueAccess(venue, handleAccessDenied);
 
-  if (venueRequestStatus && !venue) {
+  if (!space || !venueId) {
```

Comment:
This change here is incorrect and will stuck at infinite loading screen when the user is not logged in.

However, moving it below the `NotFound` check, won't work as well. Because that screen flashes for a second before the space is rendered.

Currently thinking of a fix for this.